src/wakefs/fuse.cpp

```diff
@@ -127,6 +129,32 @@ static bool collect_result_metadata(const std::string daemon_output, const struc
   return !result_ss.fail();
 }
 
+static void exec_in_pidns(pidns_args *nsargs) {
+  size_t stack_size = 1024 * 1024;
+  uint8_t *child_stack = (uint8_t *)malloc(stack_size) + stack_size;
+
+  // Create a new thread in a PID namespace, calling pidns_init.
+  pid_t child_pid = clone(pidns_init, child_stack, CLONE_NEWPID | SIGCHLD, nsargs);
```

Comment:
Ok I had to go read the clone docs. I knew it was how pthread_create and fork (and exec etc...) were all implemented but I didn't know the details of how to actually do that. Since neither CLONE_VM nor CLONE_THREAD is called, this is basically a fancy fork call. That also means that we don't have to worry about avoiding using TLS which was one of my first comments on this before I read more. Does that all sound correct to you?

Reply:
I think I need to wait on the child regardless, so that we can capture it's exit code